Counterexamples/Weierstrass.lean

```diff
@@ -0,0 +1,296 @@
+/-
+Copyright (c) 2025 Weiyi Wang. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Weiyi Wang
+-/
+import Mathlib.Analysis.Real.Pi.Bounds
+import Mathlib.Topology.Algebra.InfiniteSum.TsumUniformlyOn
+
+/-!
+# Weierstrass function: a function that is continuous everywhere but differentiable nowhere
+
+This file defines the real-valued Weierstrass function as
+
+$$f(x) = \sum_{n=0}^\infty a^n \cos (b^n\pi x)$$
+
+and prove that it is continuous everywhere but differentiable nowhere for $a \in (0, 1)$, and
+positive odd integer $b$ such that
+
+$$ab > 1 + \frac{3}{2}\pi$
+
+which is the original bound given by Karl Weierstrass. There is a better bound $ab \ge 1$ given by
+G. H. Hardy, but it is not implemented here.
+
+## References
+
+* [Weierstrass, Karl, *Über continuirliche Functionen eines reellen Arguments, die für keinen Werth
+des letzeren einen bestimmten Differentialquotienten besitzen*][weierstrass1895]
+
+-/
+
+open Real Topology Filter
+
+/-!
+### Definition
+
+For real parameter $a$ and $b$, define the Weierstrass function as
+$$f(x) = \sum_{n=0}^\infty a^n \cos (b^n\pi x)$$
+-/
+
+noncomputable
+def weierstrass (a b x : ℝ) := ∑' n, a ^ n * cos (b ^ n * π * x)
+
+/-!
+### Continuity
+
+We show that for $a \in (0, 1)$, the summation in the definition is uniformly convergent,
+each term is uniformly continuous, and therefore Weierstrass function is uniformly continuous.
+-/
+
+theorem hasSumUniformlyOn_weierstrass {a : ℝ} (ha : a ∈ Set.Ioo 0 1) (b : ℝ) :
+    HasSumUniformlyOn (fun n x ↦ a ^ n * cos (b ^ n * π * x)) (weierstrass a b) Set.univ := by
+  have habs : |(|a|)| < 1 := by grind [abs_abs, abs_lt]
+  apply HasSumUniformlyOn.of_norm_le_summable (summable_geometric_of_abs_lt_one habs)
+  intro n x _
+  simpa using mul_le_mul_of_nonneg_left (abs_cos_le_one (b ^ n * π * x)) (abs_nonneg (a ^ n))
+
+theorem tendstoUniformly_weierstrass {a : ℝ} (ha : a ∈ Set.Ioo 0 1) (b : ℝ) :
+    TendstoUniformly (fun s x ↦ ∑ n ∈ s, a ^ n * cos (b ^ n * π * x))
+    (weierstrass a b) atTop := by
+  rw [← tendstoUniformlyOn_univ]
+  exact (hasSumUniformlyOn_iff_tendstoUniformlyOn).mp (hasSumUniformlyOn_weierstrass ha b)
```

Comment:
We have an alias for this direction of the iff.
```suggestion
  exact (hasSumUniformlyOn_weierstrass ha b).tendstoUniformlyOn 
```
Although if you wanted to also add (in the appropriate file) `alias ⟨TendstoUniformlyOn.tendstoUniformly, _⟩ := tendstoUniformlyOn_univ`, then I wouldn't complain. Then you could write the entire proof as `(hasSumUniformlyOn_weierstrass ha b).tendstoUniformlyOn.tendstoUniformly`